src/main/java/org/example/menu/GameShopMenu.java

```diff
@@ -0,0 +1,130 @@
+package org.example.menu;
+
+import org.example.model.Game;
+import org.example.service.GameService;
+import org.example.service.UserService;
+import org.example.service.UserSingleton;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Scanner;
+
+public class GameShopMenu {
+
+    private final Scanner scanner;
+    private final GameService gameService;
+    private final UserService userService;
+    private final UserSingleton userSingleton;
+    private final UserMenu userMenu;
+    private final WalletMenu walletMenu;
+    private boolean isOn = true;
+
+    public GameShopMenu(Scanner scanner, GameService gameService, UserService userService, WalletMenu walletMenu, UserMenu userMenu) {
+        this.scanner = scanner;
+        this.gameService = gameService;
+        this.userService = userService;
+        this.userSingleton = UserSingleton.getInstance();
+        this.walletMenu = walletMenu;
+        this.userMenu = userMenu;
+    }
+
+    public void run() {
+        while (isOn) {
+            System.out.println(Messages.SHOW_SHOP_MENU);
+            switch (menuResponse()) {
+                case "1" -> showAllGames();
+                case "2" -> purchaseGameMenu();
+                case "3" -> showUserGameCollection();
+                case "4" -> System.out.println(Messages.CURRENT_BALANCE + " " + walletMenu.showBalance() + "$.");
+                case "5" -> walletMenu.recharge();
+                case "0" -> {
+                    isOn = false;
+                    userMenu.exit();
+                }
+            }
+        }
+    }
+
+    private String menuResponse() {
+        List<String> correctResponse = List.of("1", "2", "3", "4", "5", "0");
+        String menuResponse = scanner.next();
+        while (!checkResponse(menuResponse, correctResponse)) {
+            System.err.println(Messages.WRONG_OPTION);
+            menuResponse = scanner.next();
+        }
+        return menuResponse;
+    }
+
+    private boolean checkResponse(String gameResponse, List<String> correctResponse) {
+        return correctResponse.stream().
+                anyMatch(element -> element.equals(gameResponse));
+    }
+    private void showAllGames() {
+        gameService.showAll().forEach(System.out::println);
+    }
+
+    private void purchaseGameMenu() {
+        String gameResponse;
+        List<String> correctResponse = new ArrayList<>();
+        for (int i = 1; i <= gameService.getNumberOfGames(); i++) {
+            correctResponse.add(String.valueOf(i));
+        }
+        showAllGames();
+        System.out.println(Messages.CHOOSE_GAME_NUMBER);
```

Comment:
why not id? you can show games ids and don't do 67 - 70

Reply:
> why not id? you can show games ids and don't do 67 - 70

Yes, I can, but I wrote a method "checkResponse" which uses few times and if I will use findGameById I need to change logic(cause this method return Game) or write one more method. But I will remember that for the future.